src/components/Modal/index.js

```diff
@@ -47,29 +41,94 @@ class RideModal extends React.Component {
     //TODO: PUT new ride details
     this.setState({
       editing: false,
+      trip: this.state.form,
+    });
+  }
+
+  cancel() {
+    this.setState({
+      editing: false,
+      form: this.state.trip,
+    });
+  }
+
+  handleFormChange(value) {
+    this.setState({
+      form: {
+        ...this.state.form,
+        ...value,
+      },
     });
   }
 
   render() {
-    const { trip, scrollbox, driver, visible } = this.props;
-    const { form, editing } = this.state;
+    const { scrollbox, driver } = this.props;
+    const { trip, form, editing } = this.state;
+
     return (
-      <Modal open={visible} onClose={this.closeModal}>
+      <Modal onClose={this.closeModal} open={true}>
         <h2>
           {trip.start_city} to {trip.end_city}
         </h2>
-        <h4> ${trip.price_per_seat} per seat </h4>
-        <h4> {trip.seats} seats available </h4>
-        <p> {trip.info} </p>
+        {editing ? (
+          <div>
+            {/* TODO: add icons before input fields */}
+            <Input
+              label="Price per seat"
+              defaultValue={form.price_per_seat}
+              onChange={(e, value) =>
+                this.handleFormChange({ price_per_seat: value })
```

Comment:
May be easier to use linkState(). Check out register.js to see how that form works

Reply:
oh yea will do